File: src/microservices/_old_c#/events/EventModels.cs
using System.Text.Json.Serialization;

namespace EventsService.Models
{
    // Базовый класс для всех событий
    public abstract class BaseEvent
    {
        public Guid EventId { get; set; } = Guid.NewGuid();
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
        public abstract string EventType { get; }
    }

    // Событие фильма (соответствует /api/events/movie)
    public class MovieEvent : BaseEvent
    {
        [JsonIgnore]
        public override string EventType => "Movie";
        
        public int MovieId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Action { get; set; } = string.Empty; // e.g., "viewed", "updated"
        public int? UserId { get; set; } // Кто совершил действие
    }

    // Событие пользователя (соответствует /api/events/user)
    public class UserEvent : BaseEvent
    {
        [JsonIgnore]
        public override string EventType => "User";

        public int UserId { get; set; }
        public string Action { get; set; } = string.Empty; // e.g., "registered", "logged_in"
        public string? Username { get; set; }
        public string? Email { get; set; }
    }

    // Событие платежа (соответствует /api/events/payment)
    public class PaymentEvent : BaseEvent
    {
        [JsonIgnore]
        public override string EventType => "Payment";

        public int PaymentId { get; set; }
        public int UserId { get; set; }
        public decimal Amount { get; set; }
        public string Status { get; set; } = string.Empty; // e.g., "completed", "failed"
    }

    // Ответ API
    public class EventResponse
    {
        public string Status { get; set; } = "success";
        public int Partition { get; set; }
        public long Offset { get; set; }
        public BaseEvent Event { get; set; }
    }
}

File: src/microservices/events/internal/models/event_models.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// BaseEvent - Базовый класс для всех событий
type BaseEvent struct {
	EventID   uuid.UUID `json:"eventId"`
	Timestamp time.Time `json:"timestamp"`
}

type Eventer interface {
	GetEventType() string
}

// MovieEvent - Событие фильма
type MovieEvent struct {
	BaseEvent
	MovieID int    `json:"movieId"`
	Title   string `json:"title"`
	Action  string `json:"action"` // e.g., "viewed", "updated"
	UserID  *int   `json:"userId,omitempty"` // Nullable int
}

func (e *MovieEvent) GetEventType() string {
	return "Movie"
}

// UserEvent - Событие пользователя
type UserEvent struct {
	BaseEvent
	UserID   int     `json:"userId"`
	Action   string  `json:"action"` // e.g., "registered", "logged_in"
	Username *string `json:"username,omitempty"`
	Email    *string `json:"email,omitempty"`
}

func (e *UserEvent) GetEventType() string {
	return "User"
}

// PaymentEvent - Событие платежа
type PaymentEvent struct {
	BaseEvent
	PaymentID int     `json:"paymentId"`
	UserID    int     `json:"userId"`
	Amount    float64 `json:"amount"` // Используем float64 для decimal
	Status    string  `json:"status"` // e.g., "completed", "failed"
}

func (e *PaymentEvent) GetEventType() string {
	return "Payment"
}

// EventResponse - Ответ API
type EventResponse struct {
	Status    string      `json:"status"`
	Partition int32       `json:"partition"`
	Offset    int64       `json:"offset"`
	Event     interface{} `json:"event"` // Храним общее событие
}

// NewBaseEvent инициализирует базовые поля
func NewBaseEvent() BaseEvent {
	return BaseEvent{
		EventID:   uuid.New(),
		Timestamp: time.Now().UTC(),
	}
}